### PR TITLE
use HTTPS on ignition download

### PR DESCRIPTION
**Pull request by SANJEEV-Choubey**

### CODE REVIEW

File: ignition-server/controllers/machineconfigserver_ignitionprovider.go [hunk 1]
```diff
@@ -478,3 +527,25 @@ func compress(content []byte) ([]byte, error) {
 	}
 	return b.Bytes(), nil
 }
+
+func generateCert(mcsService *corev1.Service) error {
```
Comment (relyt0925) [inline ignition-server/controllers/machineconfigserver_ignitionprovider.go:531]: you should not need to generate certs here:
Certs are generated apriori in the pki function

Comment (relyt0925) [reply ignition-server/controllers/machineconfigserver_ignitionprovider.go:531]: You can use a wildcard to cover all cases AKA
`*.machine-config-server.NAMESPACE.svc.cluster.local`

Comment (relyt0925) [reply ignition-server/controllers/machineconfigserver_ignitionprovider.go:531]: You will need to ensure that fomat is added here though:
https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go#L13

File: ignition-server/controllers/machineconfigserver_ignitionprovider.go [hunk 2]
```diff
@@ -97,19 +107,23 @@ func (p *MCSIgnitionProvider) GetPayload(ctx context.Context, releaseImage strin
...
+		return nil, fmt.Errorf("failed to create machine config server service: %w", err)
+	}
+
 	mcsPod = machineConfigServerPod(p.Namespace, img, mcsServiceAccount,
 		mcsConfigConfigMap)
 	if err := p.Client.Create(ctx, mcsPod); err != nil {
 		return nil, fmt.Errorf("failed to create machine config server Pod: %w", err)
 	}
-
+	log.Println("pod created", mcsPod.Name)
```
Comment (relyt0925) [inline ignition-server/controllers/machineconfigserver_ignitionprovider.go:119]: just a note: before actual review clean up all debug statements.

File: ignition-server/controllers/machineconfigserver_ignitionprovider.go [hunk 3]
```diff
@@ -32,7 +36,7 @@ const (
 // MachineConfigServer pods to build ignition payload contents
 // out of a given releaseImage and a config string containing 0..N MachineConfig yaml definitions.
 type MCSIgnitionProvider struct {
-	Client          client.Client
+	Client          ctrlclient.Client
```
Comment (relyt0925) [inline ignition-server/controllers/machineconfigserver_ignitionprovider.go:39]: we shouldn't change this

File: ignition-server/controllers/machineconfigserver_ignitionprovider.go [hunk 4]
```diff
@@ -121,37 +135,45 @@ func (p *MCSIgnitionProvider) GetPayload(ctx context.Context, releaseImage strin
...
-		// Send proxy request.
-		client := &http.Client{
-			Timeout: 5 * time.Second,
+		//Generate Cert
+		if err := generateCert(mcsService); err != nil {
+			return false, fmt.Errorf("error generating cert for machine config server service: %w", err)
 		}
-		res, err := client.Do(proxyReq)
+
+		tlsConf := &tls.Config{}
```
Comment (relyt0925) [inline ignition-server/controllers/machineconfigserver_ignitionprovider.go:148]: we should be using the basic http client and configuring the TLS parameter of it apropriately versus trying to read over the wire

Comment (relyt0925) [reply ignition-server/controllers/machineconfigserver_ignitionprovider.go:148]: See this simple program for a example:
https://github.com/jcbsmpsn/golang-https-example/blob/master/https_client.go#L28

Comment (relyt0925) [reply ignition-server/controllers/machineconfigserver_ignitionprovider.go:148]: CA cert to use in `rootCAs` is available at the `mcs-crt` secret. Specifically the `ca.crt` field of it

File: ignition-server/controllers/machineconfigserver_ignitionprovider.go [hunk 5]
```diff
@@ -121,31 +124,48 @@ func (p *MCSIgnitionProvider) GetPayload(ctx context.Context, releaseImage strin
...
+		}
+		if tlsKey, ok = machineConfigServerCert.Data["tls.key"]; !ok {
+			return false, fmt.Errorf("failed to get tls key: %w", err)
+		}
+		cert, err := tls.X509KeyPair(tlsCert, tlsKey)
 		if err != nil {
-			return false, fmt.Errorf("error building http request for machine config server pod: %w", err)
+			return false, fmt.Errorf("failed to load cert: %w", err)
 		}
-		// We pass expected Headers to return the right config version.
```
Comment (enxebre) [inline ignition-server/controllers/machineconfigserver_ignitionprovider.go:130]: why is this dropping this headers? how would this work without them?

File: ignition-server/controllers/machineconfigserver_ignitionprovider.go [hunk 6]
```diff
@@ -331,8 +354,8 @@ cat /tmp/custom-config/base64CompressedConfig | base64 -d | gunzip --force --std
 					},
 					Ports: []corev1.ContainerPort{
 						{
-							Name:          "http",
-							ContainerPort: 8080,
+							Name:          "https",
+							ContainerPort: 8443,
```
Comment (enxebre) [inline ignition-server/controllers/machineconfigserver_ignitionprovider.go:358]: why is this changing? ins't the pod container still listening on 8080?

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 7]
```diff
@@ -21,10 +21,11 @@ import (
 	"crypto/x509"
 	"crypto/x509/pkix"
 	"fmt"
-	"k8s.io/apimachinery/pkg/api/resource"
 	"strings"
 	"time"
 
+	"k8s.io/apimachinery/pkg/api/resource"
+
```
Comment (enxebre) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:28]: please remove additional space and run go imports

File: ignition-server/controllers/machineconfigserver_ignitionprovider.go [hunk 8]
```diff
@@ -95,29 +97,24 @@ func (p *MCSIgnitionProvider) GetPayload(ctx context.Context, releaseImage strin
...
 	if err := p.Client.Create(ctx, mcsConfigConfigMap); err != nil {
 		return nil, fmt.Errorf("failed to create machine config server RoleBinding: %w", err)
 	}
-
 	mcsPod = machineConfigServerPod(p.Namespace, img, mcsServiceAccount,
 		mcsConfigConfigMap)
 	if err := p.Client.Create(ctx, mcsPod); err != nil {
 		return nil, fmt.Errorf("failed to create machine config server Pod: %w", err)
 	}
-
```
Comment (enxebre) [inline ignition-server/controllers/machineconfigserver_ignitionprovider.go:113]: why these removals?

File: ignition-server/controllers/machineconfigserver_ignitionprovider.go [hunk 9]
```diff
@@ -129,31 +126,56 @@ func (p *MCSIgnitionProvider) GetPayload(ctx context.Context, releaseImage strin
...
 		// https://www.iana.org/assignments/media-types/application/vnd.coreos.ignition+json
 		// https://github.com/coreos/ignition/blob/0cbe33fee45d012515479a88f0fe94ef58d5102b/internal/resource/url.go#L61-L64
 		// https://github.com/openshift/machine-config-operator/blob/9c6c2bfd7ed498bfbc296d530d1839bd6a177b0b/pkg/server/api.go#L269
 		proxyReq.Header.Add("Accept", "application/vnd.coreos.ignition+json;version=3.1.0, */*;q=0.1")
-
-		// Send proxy request.
-		client := &http.Client{
-			Timeout: 5 * time.Second,
-		}
+		//res, err := client.Get(fmt.Sprintf("https://%s.machine-config-server.%s.svc.cluster.local:8443", mcsPod.Name, p.Namespace))
```
Comment (enxebre) [inline ignition-server/controllers/machineconfigserver_ignitionprovider.go:171]: please remove comment.

**Comment by enxebre** [comment]
please squash commits into a single one.
/ok-to-test

**Comment by SANJEEV-Choubey** [comment]
Created a fresh PR for same - https://github.com/openshift/hypershift/pull/453

**Pull request closed without merging**